uberfire-nio2-backport/uberfire-nio2-impls/uberfire-nio2-jgit/src/main/java/org/uberfire/java/nio/fs/jgit/util/commands/SyncRemote.java

```diff
@@ -76,6 +76,9 @@ public Optional execute() throws InvalidRemoteException {
             remoteBranches.removeAll(localBranches);
 
             for (final String branch : remoteBranches) {
+                if (branch.equals("HEAD")) {
```

Comment:
You might want to use `org.eclipse.jgit.lib.Constants.HEAD` here. Just be aware we had a discussion about skipping branch creation for HEAD in [this PR](https://github.com/kiegroup/appformer/pull/243). For some reason @ederign advised that we should not skip it. Anyway it might be good idea to add comment about why we're skipping HEAD here (what problems it causes when not skipped).

Reply:
Okay, I'll definitely change the code to use the jgit constant. @ederign can you explain why we wouldn't want to filter out the HEAD here? The reason I added this here is that without this line, cloned repositories would show the "HEAD" branch in the workbench UI.

Reply:
I didn't read yet the whole thread, but my concern with removing the HEAD branch is that we could potentially lose the info if the HEAD points to another branch then master.

If the UI shows the HEAD branch I think is another issue.

Reply:
I partially agree with you @mbarkley. 

My only concern for remove the HEAD branch will be when we will sync with an  repo/project. (something that we will probably add in future)

If a repo HEAD points to another branch/commit after they do the SyncRemote command, if we compare the local with remote will have potential differences.

I've two use cases for this:

- You want to fork my repo in the current branch in order to help me with some issue; (if we don't keep track of the head, the user will have to switch from master to this branch)
- Another option for this use case would be moving wb state to a detached head and ask maven to do a build from the detached head in order to try something. (How can you inform maven that he should build from a detached head?)

For sure this could be addressed in future but I only want to highlight this with you.

I'm good with both options, it's your call.

Reply:
just to make it clear, I'm happy to merge this "as is".